components/tidb_query/src/rpn_expr/impl_math.rs

```diff
@@ -2,8 +2,72 @@ use tidb_query_codegen::rpn_fn;
 
 use crate::codec::data_type::*;
 use crate::codec::{self, Error};
+use crate::expr::EvalContext;
 use crate::Result;
 
+#[rpn_fn(capture = [ctx])]
+pub fn ceil<C: Ceil>(ctx: &mut EvalContext, arg: &Option<C::Input>) -> Result<Option<C::Output>> {
```

Comment:
It is quite common that a function taking a null always returns null. Can we just abstract this behavior?

Reply:
Maybe @andylokandy can work on this later.

Reply:
> Maybe @andylokandy can work on this later.

Should we just use a macro like `try_opt`?